fendermint/eth/api/src/apis/eth.rs

```diff
@@ -640,10 +640,10 @@ where
     let msghash = et::TxHash::from(ethers_core::utils::keccak256(rlp.as_raw()));
     tracing::debug!(?sighash, eth_hash = ?msghash, ?tx, "received raw transaction");
 
-    if let Some(tx) = tx.as_eip1559_ref() {
-        let tx = from_eth::to_eth_transaction(tx.clone(), sig, msghash);
-        data.tx_cache.insert(msghash, tx);
-    }
+    data.tx_cache.insert(
+        msghash,
+        from_eth::to_eth_transaction_response(&tx, sig, msghash)?,
```

Comment:
What do you mean with transaction _response_ here? This module is an API, so I interpret as if this is returning the response the API sends to the caller, which isn't the case? (unless there's an error)

We are duplicating the transaction type validation in this method and when we calculate the origin kind above. Can we refactor this to make it DRY?

BTW -- how can we have a module called `from_eth` that then has `to_eth` methods? Wouldn't they cancel each other out? 😆